From 7a3ddf29ce1d1e01c193789b2279933d49ba1030 Mon Sep 17 00:00:00 2001
From: Naveen S <quic_naves@quicinc.com>
Date: Wed, 31 Jul 2024 10:40:38 +0530
Subject: [PATCH] hostapd: Assign unique sta id for each sta in the mbssid

If mbssid is enabled, the sta_aid space will be shared by all the vifs
in the mbss. With this change, each sta in the mbssid will get a unique
sta aid irrespective of the bss. This can resolve the trigger frame
failure which is observed in fw.

Patch-dependency: none
Patch-work: none

Signed-off-by: Hari Chandrakanthan <quic_haric@quicinc.com>
Signed-off-by: Naveen S <quic_naves@quicinc.com>
---
 src/ap/ieee802_11.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index 09696b4..35affa9 100644
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -3513,6 +3513,8 @@ int hostapd_get_aid(struct hostapd_data *hapd, struct sta_info *sta)
 	if (TEST_FAIL())
 		return -1;
 
+	hapd = hostapd_mbssid_get_tx_bss(hapd);
+
 	for (i = 0; i < AID_WORDS; i++) {
 		u32 aid_word = hostapd_get_aid_word(hapd, sta, i);
 
-- 
2.34.1

